### PR TITLE
LEAN-3182 - Add supplements to ProseMirror doc

### PR DESCRIPTION
**Pull request by hhusban**

### CODE REVIEW

**Review by mnatsheh — APPROVED** [review]
Let's use "supplement" to be consistent with the json-schema. so, we should use `SupplementNode`, `SupplementsNode`, etc.

File: src/schema/nodes/supplements.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+interface Attrs {
+  id: string
+}
+
+export interface SupplementsNode extends ManuscriptNode {
+  attrs: Attrs
+}
+
+export const supplements: NodeSpec = {
+  content: 'section_title supplement*',
```
Comment (asouqi) [inline src/schema/nodes/supplements.ts:30]: @mnatsheh is there going to be a view for supplements in the editor!

**Comment by mbartenev-atypon** [comment]
@mnatsheh @hhusban I picked up that change in the body editor and I see that supplements now are inserted at the beginning of the document right after the keywords. Is that expected?

**Comment by mnatsheh** [comment]
This is not expected. We will include the fix for it in LEAN-3181.

**Pull request merged**